thrivecoin/sources/reward.move

```diff
@@ -0,0 +1,166 @@
+module thrivecoin::reward {
+  // imports
+  use sui::object::{Self, UID};
+  use sui::transfer;
+  use sui::tx_context::{Self, TxContext};
+  use sui::vec_set::{Self, VecSet};
+  use sui::table::{Self, Table};
+  use sui::sui::SUI;
+  use sui::coin::{Self, Coin};
+  use sui::balance::{Self, Balance};
+  use sui::types;
+
+  // errors
+  const ENotOneTimeWitness: u64 = 1;
+  const ENotWriter: u64 = 2;
+  const ETreasuryInsufficient: u64 = 3;
+  const EBalInsufficient: u64 = 4;
+
+  // structures
+  struct AdminRole has key {
```

Comment:
yes this was intentional as we want to have at most 1 admin role like owner of module, and only allow transfer within module pub function transfer_admin_role

Reply:
Re `Additionally, wrapping it in another object is not an option.`, we wanted to have only one address to own this object, would this work with multisigs? I'ts not clear to us how multisig would work

Reply:
Since you're aware it's not transferable, feel free to proceed. It should work fine! Of cource, if you want, you can use multisig. Just create the multisig address (here you can find more info about [multisig](https://docs.sui.io/concepts/cryptography/transaction-auth/multisig) and [multisig auth](https://docs.sui.io/guides/developer/cryptography/multisig)) and transfer the `AdminRole` object to it, similar to how you transferred it to another address.